CompilerSource/backend/GameData.cpp

```diff
@@ -0,0 +1,411 @@
+};
+
+SpriteData::SpriteData(const ::Sprite &sprite):
+  name(sprite.name) {
+  set_id(sprite.id);
+
+  // Discarded: sprite.transparent
+  // Discarded: sprite.maskShapes
+
+  set_shape((buffers::resources::Sprite_Shape) sprite.shape);  // Enum correspondence guaranteed
+  set_alpha_tolerance(sprite.alphaTolerance);
+  set_separate_mask(sprite.separateMask);
+  set_smooth_edges(sprite.smoothEdges);
+  set_preload(sprite.preload);
+  set_origin_x(sprite.originX);
+  set_origin_y(sprite.originY);
+
+  set_bbox_mode((buffers::resources::Sprite_BoundingBox) sprite.bbMode);  // Enum correspondence guaranteed
+  set_bbox_left(sprite.bbLeft);
+  set_bbox_right(sprite.bbRight);
+  set_bbox_top(sprite.bbTop);
+  set_bbox_bottom(sprite.bbBottom);
+
+  for (int i = 0; i < sprite.subImageCount; ++i)
+    image_data.emplace_back(sprite.subImages[i].image);
+}
+
+SoundData::SoundData(const ::Sound &sound):
+  name(sound.name),
+  audio(sound.data, sound.data + sound.size) {
+  set_id(sound.id);
+
+  // Discarded: sound.chorus
+  // Discarded: sound.echo
+  // Discarded: sound.flanger
+  // Discarded: sound.gargle
+  // Discarded: sound.reverb
+
+  set_kind((buffers::resources::Sound_Kind) sound.kind);
+  set_file_extension(sound.fileType);
+  set_file_name(sound.fileName);
+  set_volume(sound.volume);
+  set_pan(sound.pan);
+  set_preload(sound.preload);
+}
+
+BackgroundData::BackgroundData(const ::Background &background):
+  name(background.name),
+  image_data(background.backgroundImage) {
+  set_id(background.id);
+
+  legacy_transparency = background.transparent;
+
+  // Discarded: background.smoothEdges
+
+  set_preload(background.preload);
+  set_use_as_tileset(background.useAsTileset);
+  set_tile_width(background.tileWidth);
+  set_tile_height(background.tileHeight);
+  set_horizontal_offset(background.hOffset);
+  set_vertical_offset(background.vOffset);
+  set_horizontal_spacing(background.hSep);
+  set_vertical_spacing(background.vSep);
+}
+
+FontData::FontData(const ::Font &font):
+  name(font.name) {
+  set_id(font.id);
+
+  set_font_name(font.fontName);
+  set_size(font.size);
+  set_bold(font.bold);
+  set_italic(font.italic);
+
+  for (int i = 0; i < font.glyphRangeCount; ++i) {
+    normalized_ranges.emplace_back(font.glyphRanges[i]);
+    for (const auto &glyph : normalized_ranges.back().glyphs) {
+      *add_glyphs() = glyph.metrics;
+    }
+  }
+}
+
+FontData::NormalizedRange::NormalizedRange(const ::GlyphRange &range):
+    min(range.rangeMin), max(range.rangeMax) {
+  const int gcount = range.rangeMax - range.rangeMin + 1;
+  glyphs.reserve(gcount);
+  for (int i = 0; i < gcount; ++i) {
+    glyphs.emplace_back(range.glyphs[i]);
+  }
+}
+
+FontData::GlyphData::GlyphData(const ::Glyph &glyph):
+    ImageData(glyph.width, glyph.height, glyph.data,
+              glyph.width * glyph.height) { // Glyph images are alpha-only.
+  metrics.set_origin(glyph.origin);
+  metrics.set_baseline(glyph.baseline);
+  metrics.set_advance(glyph.advance);
```

Comment:
need   
'''metrics.set_width(glyph.width);
 metrics.set_height(glyph.height); ''' to fix fonts

Reply:
@gfundies The correct symbol is the [backtick key](https://en.wikipedia.org/wiki/Grave_accent#As_surrogate_of_apostrophe_or_(opening)_single_quote), not a single quote, for code blocks in both Git Markdown and Discord.